From 9775108b81e5ca9cff27145743474caaa6756f40 Mon Sep 17 00:00:00 2001
Message-Id: <9775108b81e5ca9cff27145743474caaa6756f40.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Mon, 17 Nov 2014 16:33:29 +0530
Subject: [PATCH 75/75] MUST_REBASE [VPG]: adf/intel/vlv: update the reserved
 reg bit 2

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-3547
Change-Id: I36ffa976d8d4f95ffaaa797f3f0c07580dc3aaf7
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Reviewed-on: https://vpg-git.iind.intel.com/5559
Reviewed-by: Kumar, Shobhit <shobhit.kumar@intel.com>
Tested-by: Kumar, Shobhit <shobhit.kumar@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_sp_plane.c    |   12 ++++++++++++
 .../video/adf/intel/include/core/intel_dc_config.h |    1 +
 .../video/adf/intel/include/core/vlv/vlv_dc_regs.h |    1 +
 drivers/video/adf/intel/intel_adf_device.c         |    1 +
 4 files changed, 15 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
index 958aef9..faff3a4 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
@@ -540,6 +540,7 @@ static void vlv_sp_flip(struct intel_plane *planeptr, struct intel_buffer *buf,
 	struct sp_plane_regs_value *regs = &splane->ctx.regs;
 	int plane = splane->ctx.plane;
 	int pipe = splane->ctx.pipe;
+	u32 val = 0;
 
 	REG_WRITE(SPSTRIDE(pipe, plane), regs->stride);
 	REG_WRITE(SPPOS(pipe, plane), regs->pos);
@@ -551,6 +552,17 @@ static void vlv_sp_flip(struct intel_plane *planeptr, struct intel_buffer *buf,
 	REG_POSTING_READ(SPSURF(pipe, plane));
 	vlv_update_plane_status(config->pipe,
 			plane ? VLV_SPRITE2 : VLV_SPRITE1, true);
+	/* Check for reserved register bit 2 */
+	val = REG_READ(SPSURF(pipe, plane));
+	if (config->flags & INTEL_ADF_PLANE_HW_PRIVATE_1) {
+		if (!(val & PLANE_RESERVED_REG_BIT_2_ENABLE)) {
+			val |= PLANE_RESERVED_REG_BIT_2_ENABLE;
+			REG_WRITE(SPSURF(pipe, plane), val);
+		}
+	} else if (val & PLANE_RESERVED_REG_BIT_2_ENABLE) {
+		val &= ~PLANE_RESERVED_REG_BIT_2_ENABLE;
+		REG_WRITE(SPSURF(pipe, plane), val);
+	}
 
 	return;
 }
diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index 01c4c15..a6c67e7 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -148,6 +148,7 @@ struct intel_plane_config {
 	u32 src_w;
 	u32 src_h;
 	u32 zorder;
+	u32 flags;
 	u8 alpha;
 	enum intel_plane_compression compression:4;
 	enum intel_plane_blending blending:4;
diff --git a/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h b/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h
index 232c2d0..03ab284 100644
--- a/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h
+++ b/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h
@@ -3879,6 +3879,7 @@
 #define SPCONSTALPHA(pipe, plane) _PIPE(pipe * 2 + plane, _SPACONSTALPHA, _SPBCONSTALPHA)
 #define SPGAMC(pipe, plane) _PIPE(pipe * 2 + plane, _SPAGAMC, _SPBGAMC)
 
+#define	  PLANE_RESERVED_REG_BIT_2_ENABLE	(1 << 2)
 #define	SPRITE_FORCE_BOTTOM	(1<<2)
 #define	SPRITE_ZORDER_ENABLE	(1<<0)
 
diff --git a/drivers/video/adf/intel/intel_adf_device.c b/drivers/video/adf/intel/intel_adf_device.c
index 99395e3..cf2465b 100644
--- a/drivers/video/adf/intel/intel_adf_device.c
+++ b/drivers/video/adf/intel/intel_adf_device.c
@@ -245,6 +245,7 @@ static void adf_plane_to_intel_plane_config(
 	config->transform = adf_plane->transform;
 	config->pipe = intf->pipe;
 	config->zorder = zorder;
+	config->flags = adf_plane->flags;
 }
 
 static struct driver_state *driver_state_create_and_init(void)
-- 
1.7.9.5

